compiler/rustc_typeck/Cargo.toml

```diff
@@ -29,3 +29,4 @@ rustc_trait_selection = { path = "../rustc_trait_selection" }
 rustc_ty_utils = { path = "../rustc_ty_utils" }
 rustc_lint = { path = "../rustc_lint" }
 rustc_serialize = { path = "../rustc_serialize" }
+stackful = { git = "https://github.com/nbdd0121/stackful.git", branch = "dev", default-features = false, features = ["nightly"] }
```

Comment:
Stack switching isn't possible on wasm. There is a proposal for it (https://github.com/WebAssembly/stack-switching), but AFAIK no implementations yet.

Reply:
Also this would make it significantly harder to make wasm32-wasi a tier 2 target with host tools if we ever want to do that in the future.

Reply:
I suppose we can always resort to running typeck multiple times... as long as unwinding is supported.

Reply:
Nope, wasm doesn't support unwinding either. Basically all call stack operations supported by wasm at the moment are call and return. There is no stack switching, unwinding, backtrace generation or tail calls.

Reply:
I'm not sure if asyncify can be used without emscripten. Emscripten only runs in the browser and I'm not sure if it's VFS code allows writing files and hardlinking. Also emscripten isn't all that well supported by rustc I believe. For example unwinding support seems to be broken (https://github.com/rust-lang/rust/pull/95950)

Reply:
It seems that asyncify can be used with wasm-opt: https://web.dev/asyncify.

Reply:
That page doesn't give any indication on how to use asyncify without a wrapper around the asyncified wasm module which tells when to unwind and when to rewind to the exact same point from which you unwound. Said wrapper is not possible with wasi, as the wasi runtime drives execution, not your own code. I also suspect asyncify only allows a single stack.

Reply:
I have added wasm32 support to stackful. https://github.com/nbdd0121/stackful/commit/f4d36931f2cdf77ab2d06485188fa2dcd3c51526

Reply:
What is the exact invocation of asyncify you need for that?